R/plotIndividualTimeProfile.R

```diff
@@ -0,0 +1,90 @@
+#'   [`Theme`](https://www.open-systems-pharmacology.org/TLF-Library/reference/Theme.html)
+#'    object for `{tlf}` library.
+#' @param ospPlotConfiguration A `ospPlotConfiguration` object, which is an `R6`
+#'   class object that defines plot properties (like labels, axes scaling and
+#'   limits, legend position, etc.). You can create instance of this object
+#'   using `createPlotConfiguration()` function.
+#' @param xUnit,yUnit Units for x- and y-axes, respectively.
+#'
+#' @import tlf
+#'
+#' @export
+plotIndividualTimeProfile <- function(dataCombined,
+                                      ospPlotConfiguration,
+                                      xUnit = NULL,
+                                      yUnit = NULL,
+                                      tlfTheme = NULL) {
+
+  # validation -----------------------------
+
+  validateIsOfType(dataCombined, "DataCombined")
+  validateIsOfType(ospPlotConfiguration, "ospPlotConfiguration")
+
+  # data frames -----------------------------
+
+  df <- dataCombined$toDataFrame()
+  df <- ospsuite:::.unitConverter(df, xUnit, yUnit)
+  obsData <- dplyr::filter(df, dataType == "observed")
+  simData <- dplyr::filter(df, dataType == "simulated")
+
+  # TimeProfilePlotConfiguration object -----------------------------
+
+  # Create an instance of `ospInternalPlotConfiguration` class
+  ospInternalPlotConfiguration <- .createOSPInternalPlotConfiguration(ospPlotConfiguration)
+
+  # Create an instance of `TimeProfilePlotConfiguration` class
+  individualTimeProfilePlotConfiguration <- tlf::TimeProfilePlotConfiguration$new()
+
+  # Annotations
+  individualTimeProfilePlotConfiguration$labels$title$text <- ospInternalPlotConfiguration$title
+  individualTimeProfilePlotConfiguration$labels$subtitle$text <- ospInternalPlotConfiguration$subtitle
+  individualTimeProfilePlotConfiguration$labels$xlabel$text <- ospInternalPlotConfiguration$xlabel
+  individualTimeProfilePlotConfiguration$labels$ylabel$text <- ospInternalPlotConfiguration$ylabel
+
+  # Legend Configuration
+  individualTimeProfilePlotConfiguration$legend <- ospInternalPlotConfiguration$legend
+
+  # X-Axis configuration
+  individualTimeProfilePlotConfiguration$xAxis <- ospInternalPlotConfiguration$xAxis
+
+  # Y-Axis configuration
+  individualTimeProfilePlotConfiguration$yAxis <- ospInternalPlotConfiguration$yAxis
+
+  # Background configuration
+  individualTimeProfilePlotConfiguration$background <- ospInternalPlotConfiguration$background
+
+  # Configurations for aesthetics
+  individualTimeProfilePlotConfiguration$lines <- ospInternalPlotConfiguration$lines
+  individualTimeProfilePlotConfiguration$points <- ospInternalPlotConfiguration$points
+  individualTimeProfilePlotConfiguration$ribbons <- ospInternalPlotConfiguration$ribbons
+  individualTimeProfilePlotConfiguration$errorbars <- ospInternalPlotConfiguration$errorbars
+
+  # Export configuration
+  individualTimeProfilePlotConfiguration$export <- ospInternalPlotConfiguration$export
```

Comment:
The benefit of the internal `ospInternalPlotConfiguration` class is that this code block will be **identical** across all plotting functions.